packages/mason_cli/lib/src/commands/unbundle.dart

```diff
@@ -0,0 +1,111 @@
+import 'dart:convert';
+
+import 'package:args/command_runner.dart';
+import 'package:mason/mason.dart';
+import 'package:mason_cli/src/command.dart';
+import 'package:mason_cli/src/commands/bundle.dart';
+import 'package:path/path.dart' as path;
+import 'package:universal_io/io.dart';
+
+/// {@template unbundle_command}
+/// `mason unbundle` command exposes the ability to generate brick templates
+/// from bundles.
+/// {@endtemplate}
+class UnbundleCommand extends MasonCommand {
+  /// {@macro unbundle_command}
+  UnbundleCommand({Logger? logger}) : super(logger: logger) {
+    argParser
+      ..addOption(
+        'output-dir',
+        abbr: 'o',
+        help: 'Directory where to output the generated brick template.',
+        defaultsTo: '.',
+      )
+      ..addOption(
+        'type',
+        abbr: 't',
+        help: 'Type of input bundle.',
+        allowed: ['universal', 'dart'],
+        defaultsTo: 'universal',
+      );
+  }
+
+  @override
+  final String description = 'Generates a brick template from a bundle.';
+
+  @override
+  final String name = 'unbundle';
+
+  @override
+  Future<int> run() async {
+    if (results.rest.isEmpty) {
+      throw UsageException(
+        'path to the bundle must be provided',
+        usage,
+      );
+    }
+    final file = File(results.rest.first);
+    if (!file.existsSync()) {
+      throw BundleNotFoundException(file.path);
```

Comment:
Can we just use logger.err here and return ExitCode.software.code to keep CLI exceptions scoped to the CLI?

Reply:
Done, mate!